uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/keyboard/KeyDownHandlerCommon.java

```diff
@@ -35,37 +34,83 @@
     protected final GridWidget gridWidget;
     protected final HasSingletonDOMElementResource gridCell;
 
+    private final boolean isTabKeyHandled;
+    private final boolean isEnterKeyHandled;
+    private final boolean isEscapeKeyHandled;
+
     public KeyDownHandlerCommon(final GridLienzoPanel gridPanel,
                                 final GridLayer gridLayer,
                                 final GridWidget gridWidget,
                                 final HasSingletonDOMElementResource gridCell) {
+        this(gridPanel,
+             gridLayer,
+             gridWidget,
+             gridCell,
+             true,
+             true,
+             true);
+    }
+
+    public KeyDownHandlerCommon(final GridLienzoPanel gridPanel,
+                                final GridLayer gridLayer,
+                                final GridWidget gridWidget,
+                                final HasSingletonDOMElementResource gridCell,
+                                final boolean isTabKeyHandled,
+                                final boolean isEnterKeyHandled,
+                                final boolean isEscapeKeyHandled) {
         this.gridPanel = gridPanel;
         this.gridLayer = gridLayer;
         this.gridWidget = gridWidget;
         this.gridCell = gridCell;
+
+        this.isTabKeyHandled = isTabKeyHandled;
+        this.isEnterKeyHandled = isEnterKeyHandled;
+        this.isEscapeKeyHandled = isEscapeKeyHandled;
     }
 
     @Override
     public void onKeyDown(final KeyDownEvent e) {
         final int keyCode = e.getNativeKeyCode();
-        final boolean isShiftKeyDown = e.isShiftKeyDown();
+
         switch (keyCode) {
             case KeyCodes.KEY_TAB:
+                if (isTabKeyHandled) {
+                    flush(e);
+                }
+                break;
+
             case KeyCodes.KEY_ENTER:
-                gridCell.flush();
-                moveSelection(keyCode,
-                              isShiftKeyDown);
-                e.preventDefault();
+                if (isEnterKeyHandled) {
+                    flush(e);
+                }
+                break;
 
             case KeyCodes.KEY_ESCAPE:
-                gridCell.destroyResources();
-                gridPanel.setFocus(true);
-                gridLayer.batch();
+                if (isEscapeKeyHandled) {
+                    reset();
+                }
         }
 
         e.stopPropagation();
     }
 
+    private void flush(final KeyDownEvent e) {
+        final int keyCode = e.getNativeKeyCode();
+        final boolean isShiftKeyDown = e.isShiftKeyDown();
+
+        gridCell.flush();
+        moveSelection(keyCode, isShiftKeyDown);
+        e.preventDefault();
+
+        reset();
+    }
+
+    private void reset() {
+        gridCell.destroyResources();
+        gridPanel.setFocus(true);
+        gridLayer.batch();
+    }
+/**/
```

Comment:
I think we do not need this comment, am I missing something?

Reply:
@jomarko Nope, you're not missing anything IDEA's `Reformat code...` context menu is next to `Comment with Block Comment` entry and I simply selected the wrong one! Anyway, comment removed in new commit :-)